bundles/org.openhab.core/src/main/java/org/openhab/core/internal/i18n/I18nProviderImpl.java

```diff
@@ -430,4 +439,19 @@ private <T extends Quantity<T>> void addDefaultUnit(Class<T> dimension, Unit<T>
     private <T extends Quantity<T>> void addDefaultUnit(Class<T> dimension, Unit<T> unit) {
         dimensionMap.put(dimension, Map.of(SIUnits.getInstance(), unit, ImperialUnits.getInstance(), unit));
     }
+
+    @Override
+    public Unit<Currency> getSystemCurrency() {
+        return new CurrencyUnit("@", "Default Currency");
```

Comment:
Can we default the currency based on configured `Locale`? Example:

```
String currencyCode = Currency.getInstance(LocaleProvider.getLocale()).getCurrencyCode();
String currencySymbol = Currency.getInstance(LocaleProvider.getLocale()).getSymbol();
```

Also, I'm wondering if we should/can encapsulate the Java [Currency](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/Currency.html) rather than reinventing too much. This would give us both the currency code (EUR) and the symbol (€) (if any) for display purposes.

Reply:
locale based default: maybe, that‘s why it is still WIP and one of the thing I will look into.

Encapsulate `Currency`: difficult. I already tried that some time ago, but failed. I think we don‘t need to duplicate that much, we can use a lot of what is already there for UoM.

Reply:
Sure, just giving some early feedback for possibly triggering ideas. I assume that's the purpose of draft/WIP. 🙂